ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -439,8 +452,18 @@ public void setTextAlign(@Nullable String textAlign) {
 
   @ReactProp(name = ViewProps.FONT_SIZE, defaultFloat = UNSET)
   public void setFontSize(float fontSize) {
-    if (fontSize != UNSET) {
-      fontSize = (float) Math.ceil(PixelUtil.toPixelFromSP(fontSize));
+    setFontSize(fontSize, false);
+  }
+
+  public void setFontSize(float fontSize, boolean forceRecalculation) {
+    mFontSizeInput = fontSize;
+
+    if (fontSize != UNSET || forceRecalculation) {
```

Comment:
Yes `null` in JS maps to `UNSET` in Java because `defaultFloat` is `UNSET` in the `@ReactProp`.

Thanks for verifying this works.